BUG: Render empty DataFrame as empty HTML table w/o raising IndexError.

JimStearns206 opened the pull request:
- [X] closes #15953 
 - [X] 1 test added / passed
 - [X] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``
 - [ ] whatsnew entry

A one line change to add an ``if clabels`` guard to clabels enumeration.

Returns HTML string consisting of empty table.

Help requested with format and placement of whatsnew entry.

JimStearns206 committed:
BUG: Render empty DataFrame as empty HTML table w/o raising IndexError.

TomAugspurger commented on pandas/tests/io/formats/test_style.py:
```diff
@@ -103,6 +103,12 @@ def test_render(self):
         # it worked?
 
+    def test_render_empty_df(self):
+        empty_df = DataFrame()
```
You can add a link to the issue here if you want `https://github.com/pandas-dev/pandas/issues/15953`

TomAugspurger replied:
Could you also add a couple checks with

- an index but on columns
- columns but no index

It can be as simple as `DataFrame(columns=['a']).style.render()` and `DataFrame(index=['a']).style.render()`

TomAugspurger reviewed (commented):
@JimStearns206 thanks. Whatsnew can go in `doc/source/whatsnew/v0.20.2.txt`

Just a minor request for a couple more edge cases. The fix looks good though.

TomAugspurger commented:
@JimStearns206 whoops looks like there were some extra commits in there. Are you on gitter? I can walk you through rebasing if you want.

TomAugspurger commented:
Basic idea is to

```
git fetch upstream
git rebase upstream/master
# fix any conflicts that come up
git push -f origin GH15953
```

JimStearns206 committed:
TST: Test rendering of 2 empty-ish DataFrames (#15953)

DataFrame with an index but no column, and one with a column but no index.
Add entry to whatsnew.

TomAugspurger commented:
Thanks!
